### PR TITLE
[IB] support benchmark matrix

### PR DESCRIPTION
**Pull request by lanking520**
## Description ##

Supporting benchmark handle in matrix

### CODE REVIEW

**Comment by lanking520** [comment]
https://github.com/deepjavalibrary/djl-serving/actions/runs/9101724608

File: tests/integration/instant_benchmark.py [hunk 1]
```diff
@@ -174,8 +185,14 @@ def parse_raw_template(url, override_container):
...
             mul_results = multiply_template_with_vars(name, cur_result, vars)
+            # each of the replicated deployment options
             for r in mul_results.values():
-                r['awscurl'] = r['awscurl'].encode().hex()
+                replicated_awscurl = multiply_template_with_vars(
+                    '', {'awscurl': cur_result['awscurl']}, benchmark_vars)
+                for option in replicated_awscurl.keys():
+                    replicated_awscurl[option] = replicated_awscurl[option][
+                        'awscurl'].encode().hex()
+                r['awscurl'] = replicated_awscurl
```
Comment (zachgk) [inline tests/integration/instant_benchmark.py:195]: You also need to clear the benchmark_vars below

File: tests/integration/record_benchmark.py [hunk 2]
```diff
@@ -46,55 +51,61 @@
...
-        },
-    ]
+    benchmark_vars = data["benchmark_vars"] if data["benchmark_vars"] else ""
+    metric_name = lambda n: (f"lmi_{data['instance']}_{esc(data['image'])}"
+                             f"_{esc(job_name)}_{esc(benchmark_vars)}_{n}")
+    metric_data = []
+    for metric, unit in cloudwatch_report_schema.items():
+        if metric in data.keys():
+            metric_data.append({
+                'MetricName': metric_name(metric),
```
Comment (zachgk) [inline tests/integration/record_benchmark.py:100]: There is some fudging of the metric names here. We have names in the cloudwatch_report_schema, json produced by awscurl, data object, and log lines parsed in `data_from_client` from `benchmark.log`. These might differ, but we should try to get them to all match. Right now, the `benchmark.log` will differ from the cloudwatch upload and the awscurl ones, while the data object could have either

Maybe the quickest is to make the names from `benchmark.log` match the ones in the cloudwatch_report_schema/awscurl. This should align all the names. If you do this, you will also need to update the tables in the cloudwatch dashboard as the metrics they are displaying are being renamed

File: tests/integration/record_benchmark.py [hunk 3]
```diff
@@ -46,55 +51,61 @@
 
 data = {}
```
Comment (zachgk) [inline tests/integration/record_benchmark.py:52]: If we are using local `data`, maybe move this declaration down near the `if __name__ == "__main__"` part

File: tests/integration/instant_benchmark.py [hunk 4]
```diff
@@ -219,23 +236,35 @@ def machine_translation(machine_name: str):
...
+
     bash_command = [
-        'set -euo pipefail', 'echo "Start Launching container..."',
+        'set -euo pipefail',
+        'echo "Start Launching container..."',
         f"docker pull {container}",
         f"./launch_container.sh {container} $PWD/models {machine_translation(instance)}",
-        job_template['awscurl'] + " | tee benchmark.log"
     ]
+    bash_command.extend(benchmark_command)
```
Comment (zachgk) [inline tests/integration/instant_benchmark.py:267]: nit, but can we just use the same variable for benchmark_command and bash_command. Then all the parts run will be in the file in order

**Comment by lanking520** [comment]
https://github.com/deepjavalibrary/djl-serving/actions/runs/9104207983

**Review by zachgk — APPROVED** [review]

**Pull request merged**